File: youpinhui-manager-web/src/main/java/com/youpinhui/manager/controller/SpecificationController.java
package com.youpinhui.manager.controller;

import java.util.List;
import java.util.Map;

import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import com.alibaba.dubbo.config.annotation.Reference;
import com.youpinhui.entity.PageResult;
import com.youpinhui.entity.Result;
import com.youpinhui.pojo.TbSpecification;
import com.youpinhui.pojogroup.Specification;
import com.youpinhui.sellergoods.service.SpecificationService;

/**
 * controller
 * @author Leon
 *
 *	
 */
@RestController
@RequestMapping("/specification")
public class SpecificationController {

	@Reference
	private SpecificationService specificationService;
	
	/**
	 * return all data
	 * @return
	 */
	@RequestMapping("/findAll")
	public List<TbSpecification> findAll(){			
		return specificationService.findAll();
	}
	
	
	/**
	 *  return all data with pagination
	 * @return
	 */
	@RequestMapping("/findPage")
	public PageResult  findPage(int page,int rows){			
		return specificationService.findPage(page, rows);
	}
	
	/**
	 * add 
	 * @param specification
	 * @return
	 */
	@RequestMapping("/add")
	public Result add(@RequestBody Specification specification){
		try {
			specificationService.add(specification);
			return new Result(true, "add success");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "add failed");
		}
	}
	
	/**
	 * update
	 * @param specification
	 * @return
	 */
	@RequestMapping("/update")
	public Result update(@RequestBody Specification specification){
		try {
			specificationService.update(specification);
			return new Result(true, "update success");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "update failed");
		}
	}	
	
	/**
	 * find one object
	 * @param id
	 * @return
	 */
	@RequestMapping("/findOne")
	public Specification findOne(Long id){
		return specificationService.findOne(id);		
	}
	
	/**
	 * batch delete
	 * @param ids
	 * @return
	 */
	@RequestMapping("/delete")
	public Result delete(Long [] ids){
		try {
			specificationService.delete(ids);
			return new Result(true, "delete success"); 
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "delete failed");
		}
	}
	
	/**
	 * fuzzy search 
	 *  if the object is not null, get the parameter from it as the search condition.
	 * @param TbSpecification
	 * @param page
	 * @param rows
	 * @return
	 */
	@RequestMapping("/search")
	public PageResult search(@RequestBody TbSpecification specification, int page, int rows  ){
		return specificationService.findPage(specification, page, rows);		
	}
	
	@RequestMapping("/selectOptionList")
	public List<Map> selectOptionList() {
		
		return specificationService.selectOptionList();
	}
	
}

File: youpinhui-manager-web/src/main/webapp/js/controller/specificationController.js
 //Controller
app.controller('specificationController' ,function($scope,$controller,specificationService){	
	
	$controller('baseController',{$scope:$scope});//inheritance
	
	//an entity contains specification and specificationOptionList
	$scope.entity={specification:{},specificationOptionList:[]};
	
    //test case , get all the data return a Json
	$scope.findAll=function(){
		specificationService.findAll().success(
			function(response){
				$scope.list=response;
			}			
		);
	}    
	
	//pagination
	$scope.findPage=function(page,rows){			
		specificationService.findPage(page,rows).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//update the search result
			}			
		);
	}
	
	//find one object
	$scope.findOne=function(id){				
		specificationService.findOne(id).success(
			function(response){
				$scope.entity= response;					
			}
		);				
	}
	
	//save
	$scope.save=function(){				
		var serviceObject;			
		if($scope.entity.specification.id!=null){//if it contains an id
			serviceObject=specificationService.update( $scope.entity ); //update  
		}else{
			serviceObject=specificationService.add( $scope.entity  );//add
		}				
		serviceObject.success(
			function(response){
				if(response.success){
					
		        	$scope.reloadList();//reload page
				}else{
					alert(response.message);
				}
			}		
		);				
	}
	
	 
	//batch delete
	$scope.dele=function(){		
		//receive the selected id from checkbox 			
		specificationService.dele( $scope.selectIds ).success(
			function(response){
				if(response.success){
					$scope.reloadList();//reload page
					$scope.selectIds=[];
				}						
			}		
		);				
	}
	
	$scope.searchEntity={};//define a search entity
	
	//search
	$scope.search=function(page,rows){	
		
		specificationService.search(page,rows,$scope.searchEntity).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//update the search result
			}			
		);
	}

	
	
	$scope.addOption=function(){
		$scope.entity.specificationOptionList.push({});
	}

	$scope.deleTableRow=function(index){
		$scope.entity.specificationOptionList.splice(index,1);
	}


    
});	
